src/webui/static/app/tvheadend.js

```diff
@@ -1393,6 +1393,9 @@ tvheadend.toCustomDate = function(date, format) //author: meizz, improvements by
             "\%M+" : date.getMonth()+1, //month
             "\%d+" : date.getDate(),    //day
             "\%h+" : date.getHours(),   //hour
+            "\%H+" : date.getHours() % 12 || 12,   //hour (12 hour format)
+            "\%P" : (date.getHours() >= 12) ? "PM" : "AM",   //am pr pm for 12 hour format
+            "\%p" : (date.getHours() >= 12) ? "pm" : "am",   //am pr pm for 12 hour format, lower case
```

Comment:
so this is a little 'confusing' (which is to blame the original code for sure).

The 'date' manual (potentially unrelated, but for consistency important) states:
```
       %b     locale's abbreviated month name (e.g., Jan)
       %h     same as %b
       %H     hour (00..23)
       %I     hour (01..12)
       %p     locale's equivalent of either AM or PM; blank if not known
       %P     like %p, but lower case
```

As `h` 'wrong' (should have been `H`). I don't see that`I` is already used here, so i'd probably recommend you do use that instead. And while at it, make `%H` do the same as `%h` here, so we have some consistency between 'regular date' and hts.

Further more (yes, it's stupid I agree) LOWER case p, is whatever your locale dicates, and UPPER case is the lower case version.

I'm not sure if there's any other locale for AM/PM, date seems to think so; so that is a bit iffy now, as you should use a localized string here. Personally (I'm not a maintainer) I'd ignore that and just use 'am/pm' here and forget about localization, because other then Americans, it's probably not really in use and it can always be fixed later on demand :)

Reply:
I agree with your assessment, and I've updated my pull request.